### DISCUSSION TITLE
https://github.com/pytorch/ao

### DISCUSSION THREAD

**Discussion by al-swaiti** [Comfy-Org/ComfyUI#4642] [General]
trying to make this quantization for Comfyui

**Comment by blepping** [comment]
rename the issue to `spam` is kind of rude but it looks more like they were irritated duo to you reopening the issue rather than anything to related to ethnicity.

what are you actually trying to do here - you have a NF4 quantized u-net only model that you can't load using https://github.com/blepping/ComfyUI_bitsandbytes_NF4 because it only supports loading full models and not u-net only, is that correct?

asking about BnB/NF4 stuff here isn't likely to be productive because it's considered obsolete compared to https://github.com/city96/ComfyUI-GGUF (which now also supports SD1.x and SDXL models). also if you wanted a feature added to `ComfyUI_bitsandbytes_NF4` you'd want to create it in that repo, not here. due to it being considered obsolete, it's very unlikely to get integrated into ComfyUI.

note: i am just a random person who read the discussion

**Reply by al-swaiti** [reply]
you know its just supporting community they was asking ,,, but unfortunately you saw the discussing "the exist of this member it's not bad for me its bad for comfy !"
whatever can  i have your opinion here https://github.com/comfyanonymous/ComfyUI/issues/4764 @blepping

**Reply by blepping** [reply]
> you know its just supporting community they was asking ,,, but unfortunately you saw the discussing "the exist of this member it's not bad for me its bad for comfy !"

sorry, i don't really understand what you mean.

> whatever can i have your opinion here

i took a look at it since you asked. i'm not really sure why it's an issue here though - it seems like you made some changes to the code and it stopped working. without actually knowing what you did, there's really no way anyone could help you.

also, you're making changes to a different project in a different repo, so if you made an issue it should probably in the ComfyUI-GGUF repo _but_ i don't think it really belongs there either. it seems like what you're experimenting with is a pretty simple type of quantization which is probably worse than ComfyUI's built in fp8 casting and NF4 which was pretty much abandoned as obsolete. there isn't really a reason why someone who already has a better type of quantization working would be interested in that.

that doesn't mean _you_ shouldn't keep working on it though, if you're having fun and learning stuff then it's definitely worthwhile. unfortunately, creating issues in these repos isn't really the way to share it and is (in my opinion) poor etiquette which is probably why you've gotten a negative reaction to some of your posts. unfortunately it is hard to find people who are interested in our weird projects and experiments, i feel your pain because i have plenty of my own.

by the way, the warning messages in the issue you linked are almost certainly not related to your black image problem.

**Reply by al-swaiti** [reply]
i was fixed the diffusers loader but it was better to be official for users (my codes was not organized for community ) , also theres a node for that https://github.com/DenkingOfficial/ComfyUI_UNet_bitsandbytes_NF4 which work perfect ,, whatever ,,,
the code i asked u about MR @blepping  its about aplied this quantization method for comfy models , which is not easy as for diffusers !  this one https://github.com/pytorch/ao  "thanks" also here a disscusion about apply this method for diffusers https://gist.github.com/sayakpaul/e1f28e86d0756d587c0b898c73822c47 ,,,, Hi @city96 ,,, excuse me do u have any idea !

**Reply by blepping** [reply]
> the code i asked u about MR @blepping its about aplied this quantization method for comfy models , which is not easy as for diffusers !

your best bet is to use either the existing NF4 or GGUF stuff as an example of how to implement quantization in ComfyUI. like i mentioned before though, what you're trying to use is likely worse than the existing options so people probably won't be that enthusiastic about doing so.

certainly nothing stopping you from making it as a custom node though, and if you're learning something by doing so then it may be worth the time.
